### PR TITLE
STY: absolute imports in __init__ files

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/__init__.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
-from .base import (  # noqa: F401
+from pandas.core.arrays.base import (  # noqa: F401
```
Comment (WillAyd) [inline pandas/core/arrays/__init__.py]: If you wanted to clean up more I think preferable to specify __all__ instead of noqa: F401 for these

**Review by WillAyd — APPROVED** [review]
lgtm ping on green

File: pandas/core/arrays/__init__.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/core/arrays/__init__.py]: can you give me an example?
I don't understand what do you mean by __all__ or you meant \_\_all__?

**Comment by MomIsBestFriend** [comment]
I was just about to ping :)

File: pandas/core/arrays/__init__.py (hunk 1 above)
Comment (WillAyd) [reply pandas/core/arrays/__init__.py]: Yea the latter

**Comment by WillAyd** [comment]
Thanks @MomIsBestFriend

**Comment by jorisvandenbossche** [comment]
Personally, I am -1 on those changes. 
@MomIsBestFriend before you do any more of such PRs (although I don't know if there are cases left?), can we first have a discussion about this?

**Comment by MomIsBestFriend** [comment]
@jorisvandenbossche I actually did want to discuss this, so I opened [this PR](https://github.com/pandas-dev/pandas/pull/30822) which as you can see has only two changes. And when it got merged I saw it as an approval to go and change even more files.

Anyway, I don't mind reverting #30822, #30827, #30828.
Until this topic is fully discussed.

**Pull request merged**